src/stateManager/StateManager.ts

```diff
@@ -846,40 +852,75 @@ class StateManager {
 
     private async createStateSnapshot(
         stateMachineStateHash: Hash,
-        forkId: ForkId,
-        blockHeight: BlockHeight
-    ): Promise<StateSnapshot> {
-        const participants = await this.diamondStateMachine.getParticipants();
+        block: Block,
+        exitChannels?: ExitChannelStruct[]
+    ): Promise<{
+        stateSnapshot: StateSnapshot;
+        exitChannelBlock?: ExitChannelBlockStruct;
+        totalWithdrawals: BalanceStruct;
+    }> {
+        const previousStateSnapshot = this.storage.getPreviousStateSnapshot(
+            block.coordinates
+        )!;
 
         const latestJoinChannelBlockHash =
-            this.storage.joinChannelBlocks.getLatestJoinChannelBlockHash();
-        const latestExitChannelBlockHash =
-            this.storage.exitChannelBlocks.getLatestExitChannelBlockHash();
-        const totalDeposits = this.storage.joinChannelBlocks.getTotalDeposits();
-        const totalWithdrawals =
-            this.storage.exitChannelBlocks.getTotalWithdrawals();
+            (await this.localDiamondContract.getLatestJoinChannelBlockHash(
```

Comment:
this should be - `genesisSnapshot.latestJoinChannelBlockHash` - you can't join off-chain in the same fork - only on-chain which creates another fork